root/static/scripts/edit/URLCleanup.js

```diff
@@ -1488,7 +1478,39 @@ const CLEANUPS = {
           result: false,
         };
       }
-      return {result: true};
+      const m = /^https:\/\/www\.instagram\.com\/([^\/]+)\/([^\/?#]+\/)?$/.exec(url);
+      if (m) {
+        const prefix = m[1];
+        const target = m[2];
+        if (id === LINK_TYPES.streamingfree.recording) {
+          return {
+            result: prefix === 'p' && target !== undefined,
+          };
```

Comment:
Should we have any sort of error message if the user tries to link a profile or some other non /p/ link? Something about linking profile pages at the artist/label/place level or whatnot?

Reply:
I don’t think it is needed. Users that find out how to edit recordings most likely know that non /p/ links don’t make sense here.

Reply:
> Users that find out how to edit recordings most likely know that non /p/ links don’t make sense here.

That's probably less true now that this also supports releases :)

Reply:
Less true but still unlikely to happen as there were only 2 instances of non /p/ Instagram URLs linked to releases so far.